boa/src/object/internal_methods.rs

```diff
@@ -881,6 +881,68 @@ impl GcObject {
         Ok(list)
     }
 
+    /// It is used to iterate over names of object's keys.
+    ///
+    /// More information:
+    /// - [EcmaScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-enumerableownpropertynames
+    pub(crate) fn enumerable_own_property_names(
+        &self,
+        kind_key: bool,
+        kind_value: bool,
+        context: &mut Context,
+    ) -> Result<Vec<JsValue>> {
```

Comment:
Was also my initial idea, but opted out for booleans because I didn't know where to put the new enum in this project 😄 

If you can recommend where I should put it to keep your code clean them I will change it right away!

Reply:
It is related to `Property`, so I think the best place to put it would be on the `property` module.

Reply:
Ok, will make the change some time later today